### PR TITLE
Poc react redux

### PR DESCRIPTION
**Pull request by stepek**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,58 +38,53 @@
...
-    "jquery-ui": "^1.10.5",
-    "js-yaml": "^3.5.5",
-    "spinkit": "^1.2.5",
-    "underscore": "^1.8.3"
+    "axios": "^0.13.1",
+    "react": "^15.3.0",
+    "react-dom": "^15.3.0",
+    "react-redux": "^4.4.5",
+    "react-router": "^2.6.1",
+    "react-spinkit": "^1.1.11",
```
Comment (thealmightyone) [inline package.json]: okay?

File: src/app/App.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import React, {Component, PropTypes} from 'react';
```
Comment (thealmightyone) [inline src/app/App.jsx]: how about we only import React and use React.Component for readability

File: src/auth/Auth.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          }
+          return (<Spinner spinnerName='double-bounce' />);
+        })()}
+      </div>
+    );
+  }
+}
+
+Auth.contextTypes = {
+  router: React.PropTypes.object
```
Comment (thealmightyone) [inline src/auth/Auth.jsx]: Same thing,  consistency regardless of we end up using

File: package.json [hunk 4]
```diff
@@ -1,16 +1,16 @@
 {
   "name": "gohan-webui",
-  "version": "1.8.0",
+  "version": "1.6.0",
```
Comment (thealmightyone) [inline package.json]: 2.0?

File: src/auth/requireAuthentication.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    componentWillReceiveProps(nextProps) {
+      this.checkCredentials(nextProps.tokenId, nextProps.tokenExpires);
+    }
+
+    render() {
+      return (
+        <div>
+          {this.props.tokenId ? <ChildComponenet {...this.props}/> : null}
```
Comment (thealmightyone) [inline src/auth/requireAuthentication.js]: literowka

File: src/auth/requireAuthentication.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+
+export default function requireAuthentication(ChildComponenet) {
```
Comment (thealmightyone) [inline src/auth/requireAuthentication.js]: typo

File: src/schema/SchemaActions.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    const state = getState();
+    const url = state.configReducer.gohan.url + state.configReducer.gohan.schema;
+    const headers = {
+      'Content-Type': 'application/json',
+      'X-Auth-Token': state.authReducer.tokenId
+    };
+
+
+    axios.get(url, {headers}).then(response => {
+      console.log('fetched');
```
Comment (thealmightyone) [inline src/schema/SchemaActions.js]: meh

**Comment by thayashi** [comment]
Looks good so far, thanks. Let's merge this to this branch

**Comment by stepek** [comment]
Fixed all comments.

**Pull request merged**